lms_homework/No21/lesson_21.py

```diff
@@ -0,0 +1,50 @@
+import os
+
+
+class File:
+    COUNTER = 0
+
+    def __init__(self, file_name, method):
+        self.name = file_name
+        self.method = method
+
+    def __enter__(self):
+        if self.method in ("w", "a") or (
+            os.path.isfile(self.name) and self.method in "r"
+        ):
+            File.COUNTER += 1
+            log_file = open(
+                "D:/Beetroot Academy II/lms_homework/No21/log_info/log.txt", "a"
+            )
+            log_file.write(f"> Create file No{File.COUNTER}\n")
+            log_file.close()
+            self.file = open(self.name, self.method)
+            return self.file
+        raise TypeError("Wrong params")
+
+    def __exit__(self, types, value, traceback):
+        self.no_error_file = File.COUNTER + 1
+        log_file = open(
+            "D:/Beetroot Academy II/lms_homework/No21/log_info/log.txt", "a"
+        )
+        log_file.write(f"> Error of create file No {self.no_error_file}")
```

Comment:
`num_of_error_file` or `error_file_num`
`no_error_file` means there is no error file )